.circleci/config.yml

```diff
@@ -222,26 +226,51 @@ commands:
 jobs:
   api-build-test:
     <<: *java_defaults
+    parallelism: 3
     steps:
       - checkout_init_git
       - ensure_branch_has_changes:
           dir_name: "api"
       - manage_api_cache:
           restore: true
       - run:
-          name: Validate Swagger definitions
+          name: Run Java unit tests
           working_directory: ~/workbench/api
-          command: ./project.rb validate-swagger --project-prop verboseTestLogging=yes
+          # Refers to https://circleci.com/docs/2.0/language-java/#sample-configuration
+          command: |
+            cd ~/workbench/api/src/test/java/org/pmiops/workbench/api
+            CLASSNAMES=$(circleci tests glob "**/*.java" \
+              | cut -c 1- | sed 's@/@.@g' \
+              | sed 's/.\{5\}$//' \
+              | circleci tests split --split-by=timings --timings-type=classname)
+            GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests",$i}')
+            cd ~/workbench/api
+            ./project.rb test $GRADLE_ARGS
       - run:
-          name: Integration tests compile
-          working_directory: ~/workbench/api
-          command: ./project.rb gradle compile__bigquerytest__Java compile__integration__Java
+          name: Collecting test results
+          command: |
+            mkdir -p ~/workbench/api/test-results/
+            find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} ~/workbench/api/test-results/ \;
+          when: always
+      - store_test_results:
+          path: ~/workbench/api/test-results
+      - store_artifacts:
+          path: ~/workbench/api/test-results
+          destination: test-results
+      - manage_api_cache:
+          save: true
+
+  api-lint-swagger-check:
```

Comment:
I'd prefer that we move these steps back into the API test, then add a `when: always` to " Run Java unit tests". This fixes the ordering concerns that we had previously. Unless there are major latency tradeoffs here.